applicationset/generators/git.go

```diff
@@ -167,7 +168,11 @@ func (g *GitGenerator) generateParamsFromGitFile(filePath string, fileContent []
 			paramPath["basenameNormalized"] = utils.SanitizeName(path.Base(paramPath["path"].(string)))
 			paramPath["filenameNormalized"] = utils.SanitizeName(path.Base(paramPath["filename"].(string)))
 			paramPath["segments"] = strings.Split(paramPath["path"].(string), "/")
-			params["path"] = paramPath
+			if pathParamPrefix != "" {
+				params[pathParamPrefix] = map[string]interface{}{"path": paramPath}
```

Comment:
What happens if I use another parameter's name as the prefix? For example, `"revision"` - will it overwrite that parameter?

Reply:
Hrm, yeah, it would overwrite.

Is that acceptable? Or should it error here if `params[pathParamPrefix]` already exists?

Reply:
I don’t have a strong opinion on this topic but, FWIW, there’s already an implicit caveat that if your file content includes a field called `path`, it’s going to get overwritten here. I feel like, as long as the user knows what the behavior is, they can choose a prefix that won’t conflict with the expected file content.

Reply:
Makes sense! I'm okay with the overwrite behavior... Might want to add a note to the docs, just for good measure.